src/shims/time.rs

```diff
@@ -117,6 +117,38 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         Ok(())
     }
 
+    #[allow(non_snake_case)]
+    fn QueryPerformanceCounter(&mut self, lpPerformanceCount_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.assert_target_os("windows", "QueryPerformanceCounter");
+        this.check_no_isolation("QueryPerformanceCounter")?;
+
+        // QPC uses a hardware counter as its basis.
+        // Miri will assume that the machine's hardware counter has a resolution of 1 nanosecond.
+        let duration = Instant::now().duration_since(this.machine.time_anchor);
+        let qpc = i64::try_from(duration.as_nanos())
```

Comment:
Oh wow windows uses signed values for hardware performance counters? Interesting.
The cast looks correct though.

Reply:
I found an explanation (from the docs link above) on why they are using signed values.
```
Why is the time stamp that is returned from QPC a signed integer?

Calculations that involve QPC time stamps might involve subtraction. By using a signed value, you can handle calculations that might yield negative values.
```